build_test.go

```diff
@@ -197,6 +198,7 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 					},
 				},
 			}))
+		Expect(entryResolver.MergeLayerTypesCall.CallCount).To(Equal(1))
 
 		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
```

Comment:
To test that the MergeLayerTypes function is being used correctly in `build.go`, you can make assertions about the arguments passed to the function call. Like this line that tests an argument passed to `dependencyManager.Resolve()`.
```
		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
```
If you test what the entryResolver MergeLayerTypes call _receives_, you don't need to separately test its call count in this case, because if call count = 0, the `Receives` field will have its zero value. That way, one assertion checks a) `MergeLayerTypes()` is called and b) it's called with appropriate arguments.